app/src/main/java/com/neudesic/myapplication/data/repository/DadJokeRepository.kt

```diff
@@ -0,0 +1,65 @@
+package com.neudesic.myapplication.data.repository
+
+import com.neudesic.myapplication.domain.repository.DadJokeRepository
+import com.neudesic.myapplication.domain.models.DadJoke
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.GET
+import retrofit2.http.Headers
+import java.lang.Exception
+
+
+class DadJokeRepositoryImpl : DadJokeRepository {
+    override suspend fun getJoke(): Response<DadJoke>? {
+        // TODO: Extract. see https://codersee.com/retrofit-with-kotlin-the-ultimate-guide/
+        val retrofit = Retrofit.Builder()
```

Comment:
If you want to get real fancy, you can have dagger provide a singleton of this instance.

Reply:
I did read about that, but ive no idea how to set that up yet.
If you have a lead or pseudocode for me, that would be awesome :D

Reply:
I'd want the singleton to be able to have dynamic base URL if possible.

Reply:
Hmm, you could have a single Retrofit instance for each api.